### ISSUE TITLE
Tap 18.2: SyntaxError: The requested module 'minipass' does not provide an export named 'Minipass'

### ISSUE THREAD

**Issue by piotr-cz** [tapjs/tapjs#927]
After updating to tap 18.2 I'm receiving following error when trying to run tests:

```
npx tap
xxx\node_modules\@tapjs\reporter\src\json.ts:4
import { Minipass } from 'minipass'
         ^

SyntaxError: The requested module 'minipass' does not provide an export named 'Minipass'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:190:5)

Node.js v18.18.0
```

When I open file `node_modules\@tapjs\reporter\dist\esm\json.js` in VS Code and click on `minipass`, it opens up `node_modules\minipass\index.d.ts` and doesn't show any errors.

I have also noticed that `test` function doesn't provide type for it's parameter anymore:

```ts
// t: any, but should be t:Tap.Test
test('should test something', async t => {
))
```

I have no idea what's wrong, might something introduced here: https://github.com/tapjs/tapjs/compare/%40tapjs/reporter%401.1.4...%40tapjs/reporter%401.2.0

Or it may be somehow related to ts-node

**Comment by isaacs** [comment]
My guess is that you're using `"moduleResolution": "node"` or `"moduleResolution":"node10"`, instead of `"moduleResolution": "node16"` or `"moduleResolution":"nodenext"`.  In `"node10"` mode, TSC does not resolve package.json `exports`, which is where all the types of these modules' entry points are being defined.

I'm working through a backlog of setting main/types to the commonjs values everywhere to help folks who are stuck on `node10` moduleResolution for some reason, but tbh, if this is the situation you're in, the best thing to do is just tell TSC that you are using a node version that isn't 7 years out of date.

**Comment by isaacs** [comment]
OH, another common problem, which has bitten me a few times:

If you have something like this in tsconfig.json:

```json
{
  "include": ["src/**/*.ts"],
  "compilerOptions": {
    "moduleResolution": "nodenext",
    ... other stuff ...
  }
}
```

then that's telling tsc that the `compilerOptions` (includeing the package.json exports support) _only_ applies to files in `src/**/*.ts`, so tsserver (and thus, your editor) and tsnode (and thus your test run) won't properly see what's going on in your tests.

The solution that I use is to put all my compilerOptions in `./tsconfig.json`, and then have a separate `tsconfig.build.json` that contains:

```json
{
  "extends": "./tsconfig.json",
  include/exclude/outdir/etc.
}
```

and then build with `tsc -p tsconfig.build.json`.

**Comment by isaacs** [comment]
Another thing I just noticed, `Tap.Test` is actually _not_ what it should be, that's the type defined in `@types/tap`, which is not applicable to v18. You'll probably have to remove that, I would think that it shouldn't cause any problems if tsc is set up to find the types properly, but it might be making things weird.

In CommonJS mode, the type will be exported as `tap.Test` (lowercase `t`), and in ESM, it's directly exported on `import type { Test } from 'tap'`.

**Comment by isaacs** [comment]
Ok, yes, all of what I said above is valid, but the main problem here is that `@tapjs/reporter` doens't have a dependency on `minipass`, so it's getting the wrong version, because something else in your tree is giving it a version other than minipass v7. Will correct shortly.

**Comment by piotr-cz** [comment]
I can confirm that the 6a7d376 fixed this issue (tested after updating to v18.3)

As for the `t` type, I mentioned it here, because I thought it's somehow related but it's another issue.
Looks like it's not properly interfered

---
Edit: 
I've noticed that this applies to commonjs exports and when I switched compiler options in tsconfig.json

```diff
-  "module": "es2022",
-  "moduleResolution": "node",

+  "module": "node16"
+  "moduleResolution": "node16",
```

`t` became properly interfered.

I'm extending my tsconfig with https://www.npmjs.com/package/@tsconfig/node18-strictest-esm so that's where the values came from.
